src/Workspaces/CSharp/Portable/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -918,28 +920,12 @@ public static SimpleNameSyntax GetNameWithTriviaMoved(this MemberAccessExpressio
         private static void GetReplacementCandidates(
             SemanticModel semanticModel,
             MemberAccessExpressionSyntax memberAccess,
-            SymbolInfo actualSymbol,
+            ISymbol actualSymbol,
             out ImmutableArray<ISymbol> speculativeSymbols,
             out ImmutableArray<ISymbol> speculativeNamespacesAndTypes)
         {
-            bool containsNamespaceOrTypeSymbol;
-            bool containsOtherSymbol;
-            if (actualSymbol.Symbol is object)
-            {
-                containsNamespaceOrTypeSymbol = actualSymbol.Symbol is INamespaceOrTypeSymbol;
-                containsOtherSymbol = !containsNamespaceOrTypeSymbol;
-            }
-            else if (!actualSymbol.CandidateSymbols.IsDefaultOrEmpty)
-            {
-                containsNamespaceOrTypeSymbol = actualSymbol.CandidateSymbols.Any(symbol => symbol is INamespaceOrTypeSymbol);
-                containsOtherSymbol = actualSymbol.CandidateSymbols.Any(symbol => !(symbol is INamespaceOrTypeSymbol));
```

Comment:
💭 I'm a bit concerned about this change. Is it observable?

Reply:
Here's a case where it _might_ go on this code path:

```
class Example
{
    void Method()
    {
        _ = nameof(this.ToString); // Should offer to simplify to `nameof(ToString)`
    }

    public override string ToString() => throw null;
    public string ToString(string format, IFormatProvider formatProvider) => throw null;
}
```

Reply:
➡️ This issue was resolved by filing follow-up issues. The current code handled ambiguous symbols, but other parts of the simplification system did not and in the end it did not work prior to this change.

#40972
#40973